dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -383,15 +383,19 @@ HRESULT WINAPI CRecycleBinItemContextMenu::InvokeCommand(LPCMINVOKECOMMANDINFO l
         if (lpcmi->lpVerb == MAKEINTRESOURCEA(1))
         {
             /* restore file */
-            if (RestoreFile(Context.hDeletedFile))
-                return S_OK;
-            else
-                return E_FAIL;
+            BOOL ret = RestoreFile(Context.hDeletedFile);
+            
+            CloseRecycleBinHandle(Context.hDeletedFile);
+            
+            return (ret) ? S_OK : E_FAIL;
```

Comment:
```suggestion
            return (ret ? S_OK : E_FAIL);
```

Reply:
I wonder whether we could factor out the `BOOL ret`, the `CloseRecycleBinHandle` call and the return, and put that as the common return path of the main `if (lpcmi->lpVerb == MAKEINTRESOURCEA(1) || lpcmi->lpVerb == MAKEINTRESOURCEA(5))` case 🤔

Reply:
With the first, i've refactored that too, but i'm waiting or responses on other areas

Reply:
Done